File: samples/java/05wc/wc.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.io.FileNotFoundException;

public class wc {

    public static void main(String[] args) {
        if (args.length > 1) {
            System.err.println("Error: Usage 'java wc [<file>]'");
            System.exit(1);
        }
        BufferedReader br = null;
        if (args.length == 1) {
            try { 
                br = new BufferedReader(new FileReader(args[0]));
            } catch (FileNotFoundException err) {
                System.err.printf("Error: File '%s' not found.", args[0]);
                System.exit(2);
            }
        } else {
            br = new BufferedReader(new InputStreamReader(System.in));
        }
        int count = 0;
        try {
            while (br.readLine() != null) {
                count++;
            }
        } catch (IOException err) {
            err.printStackTrace();
        } finally {
            try {
                if (args.length == 1) {
                    br.close();
                }
            } catch (IOException err2) {
                System.err.printf("Failed to close %s\n", args[0]);
            }
        }
        System.out.println(count);
    }

}

File: samples/kotlin/05wc/wc.kt
import java.io.File;

fun main(args: Array<String>) {
    if (args.size > 1) {
        System.err.println("Error: Usage `kotlin wc [<file>]`");
        System.exit(1);
    }
    var count: Int = 1;
    if (args.size == 1) {
        File(args[0]).forEachLine { count++ };
    } else {
        while (readLine() != null) {
            count++;
        }
    }
    println(count);
}
